core/src/main/java/com/linecorp/armeria/common/logging/ContentPreviewerFactoryBuilder.java

```diff
@@ -0,0 +1,257 @@
+/*
+ * Copyright 2020 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.common.logging;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static java.util.Objects.requireNonNull;
+
+import java.nio.charset.Charset;
+import java.util.function.BiFunction;
+import java.util.function.BiPredicate;
+
+import javax.annotation.Nullable;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Streams;
+
+import com.linecorp.armeria.common.HttpHeaders;
+import com.linecorp.armeria.common.MediaType;
+import com.linecorp.armeria.common.RequestContext;
+import com.linecorp.armeria.common.RequestHeaders;
+import com.linecorp.armeria.common.ResponseHeaders;
+import com.linecorp.armeria.common.logging.PreviewSpec.PreviewMode;
+import com.linecorp.armeria.internal.common.ArmeriaHttpUtil;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.ByteBufUtil;
+
+/**
+ * A builder which builds a {@link ContentPreviewerFactory}.
+ */
+public final class ContentPreviewerFactoryBuilder {
+
+    //TODO(minwoox): Add setters for the seprate request and response previewer.
+
+    private final ImmutableList.Builder<PreviewSpec> previewSpecsBuilder = ImmutableList.builder();
+    private int maxLength;
+    private Charset defaultCharset = ArmeriaHttpUtil.HTTP_DEFAULT_CONTENT_CHARSET;
```

Comment:
I think we either need to 

- change this to UTF_8
- Have a special case for JSON which only supports UTF-8

I lean towards the former, I think users will have more gotchas with a default matching the ancient HTTP standard vs common practice nowadays, UTF-8 for everything.

Reply:
> the ancient HTTP standard vs common practice nowadays

Yes, it's time to throw the ancient standard away. Thanks!

Reply:
Modern clients almost always send content type with a `charset` attribute, so defaulting to ISO-8859-1 makes more sense in my opinion. The only exception is JSON, which is by definition UTF-8 if no charset is specified.

(+1 for latter therefore)

Reply:
We seem to already have a strange dichotomy

- `StringRequestConverterFunction` and `StringResponseConverterFunction` have different defaults, former is ISO-8859-1 and latter is UTF-8. I don't see any reason for this other than we feel it's tremendously more convenient for responses to default to UTF-8, but this seems to apply to requests too. Why respect HTTP RFC for requests but not responses?
  - Also an example of requests that don't have charset applied even in modern times - manual debug requests from curl

- `spring-boot` defaults to UTF-8 for both. I think that's a good enough industry standard for us to follow (actually our annotated service is not a drop-in replacement for spring if we don't)

https://docs.spring.io/spring-boot/docs/current/reference/html/appendix-application-properties.html
Search `spring.http.encoding.charset`
Confirmed in code
https://github.com/spring-projects/spring-boot/blame/711391cf2f27dd2637a1bb4d7847a2323b6c4e16/spring-boot-project/spring-boot/src/main/java/org/springframework/boot/web/servlet/server/Encoding.java#L37
https://github.com/spring-projects/spring-boot/blob/711391cf2f27dd2637a1bb4d7847a2323b6c4e16/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/http/HttpMessageConvertersAutoConfiguration.java#L81

Reply:
I guess even ourselves are confused about the defaults... :sweat_smile: Let's just default to UTF-8 then.

Reply:
Fixed. 😉